### PR TITLE
Fix for macOS Catalina

### PR DESCRIPTION
**Pull request by ShikaSD**
Asked for location permission on app start.

### CODE REVIEW

**Comment by ShikaSD** [comment]
Fixes #30 #32 #25 
Thanks to @anonymouz4 for explanation

**Comment by anonymouz4** [comment]
@ShikaSD Back then, few days after I commented I even found a better solution. The patch of not giving you the mac-addr of the found devices was implemented pretty badly, they just changed few lines in the CoreWLAN framework. It can simply be bypassed if you call the functions from the underlying Apple80211 framework yourself (exactly what CoreWLAN does). This way you can have locationservices disabled. I even created my own repo and hosted it, but I made it private after a while.

**Comment by wootwoot1234** [comment]
@anonymouz4 That's great!  Can you share the repo or make a pull request?  Thanks!

**Comment by wootwoot1234** [comment]
@ShikaSD I downloaded your repo and built it in xcode.  It did prompt me for location services and I clicked OK but it still crashed the computer.  I'm running Mojave still so maybe that's it?  Is there anything else you need to do to enable location services?

**Comment by ShikaSD** [comment]
@wootwoot1234 Tbh, I don't have Mojave around to test it, but "crashed the computer" sounds rather weird :)
I just went through the path of least resistance (requesting location), which helped me to make this tool work. Could it be some bug in macOS kernel, which crashes the system or smth?

**Comment by wootwoot1234** [comment]
I know it's weird but I'm not the only person experiencing it.  Many people list that as an issue, for example, https://github.com/unixpickle/JamWiFi/issues/24.  The rebooting is even referenced in the issues you listed as fixed.  

I'll update my OS and tray again.  Thanks!

**Comment by wootwoot1234** [comment]
@ShikaSD is there a way to manually check if the location services are actually enabled?

**Comment by anonymouz4** [comment]
@wootwoot1234 I could send you a test version if you'd like

**Comment by wootwoot1234** [comment]
Yes, that would be great!

**Comment by wootwoot1234** [comment]
@ShikaSD its still crashing on Catalina. I’m running it on a late 2018 MacBook Pro.

**Comment by anonymouz4** [comment]
@wootwoot1234 you got an email?

**Comment by wootwoot1234** [comment]
tkrones@gmail.com

**Comment by wootwoot1234** [comment]
@anonymouz4 thanks for the file but it still is crashing my computer.  I'm not sure what settings are different on my computer that is causing this issue.

**Comment by ajkblue** [comment]
It is also crashing my MacBook Pro running Catalina 10.15.2

**Comment by jesse-c** [comment]
Not crashing for me on macOS 10.15.2 (2019 MacBook Pro)

**Comment by jesse-c** [comment]
Not crashing on 10.15.2 (2019 MacBook Pro).

Thanks for your work, @ShikaSD.

**Comment by wootwoot1234** [comment]
> Not crashing on 10.15.2 (2019 MacBook Pro).
> 
> Thanks for your work, @ShikaSD.

@jesse-c Did you install any drivers or change any settings that you can think of?

**Comment by jesse-c** [comment]
@wootwoot1234 nothing that I can think of. This laptop is about 2 weeks old and I haven't done too much mucking about.

I built the project with the latest Xcode and commandline tools installed.

**Comment by ctopherwilliams** [comment]
It is crashing on my MacBook Air (Retina, 13-inch, 2018) 10.15.2 (19C57)

xcode-select version 2373

**Review by mekel1991 — APPROVED** [review]

**Comment by mekel1991** [comment]
Can anyone teach me how to apply this code in layman terms? I am new to this and I really need this program

File: JamWiFi/ANListView.m [hunk 1]
```diff
@@ -134,6 +134,7 @@ - (void)scanInBackground {
         self->interfaceName = interface.interfaceName;
         NSError * error = nil;
         NSArray * nets = [[interface scanForNetworksWithSSID:nil error:&error] allObjects];
+        NSLog(@"Nets%@", nets.firstObject);
```
Comment (RSully) [inline JamWiFi/ANListView.m]: Please remove these logs or format them nicer if you believe they should be kept in master

**Comment by anonymouz4** [comment]
Doesn't fix crashing but everything else [0x0XDev/JamWiFi](https://github.com/0x0XDev/JamWiFi)

**Comment by RSully** [comment]
@anonymouz4 

You could've cloned the repo and worked off of it instead instead of wiping the git history clean. 

It is also worth noting that this original codebase is not licensed under any open-source license.

**Comment by anonymouz4** [comment]
@RSully I actually created it for myself and had no intention of making it public, thats why I wanted to start fresh, not cloning. As you can see, I kept it private for 6 month now, but since people asked if I would share bc their version didn't work, I just tried help. If it bother you that much, I could very well make it private again

**Comment by drakenass** [comment]
> @RSully I actually created it for myself and had no intention of making it public, thats why I wanted to start fresh, not cloning. As you can see, I kept it private for 6 month now, but since people asked if I would share bc their version didn't work, I just tried to help. If it bother you that much, I could very well make it private again

i'm new to GitHub and I have no idea how to install that fix, i would appreciate any help

**Comment by drakenass** [comment]
> Doesn't fix crashing but everything else [0x0XDev/JamWiFi](https://github.com/0x0XDev/JamWiFi)

i get '' Application cannot be opened'' when ever I open the app, any idea?

**Comment by anonymouz4** [comment]
@drakenass maybe download master and build it yourself. But for issues, open one on that repo, not here

**Comment by drakenass** [comment]
@anonymouz4
It was fixed. A quick general question though concerning the app it self, why can’t I use the network after jamming it on the Mac?

**Comment by kushney** [comment]
@ShikaSD macOS Catalina 10.15.2, bssid is displayed, on second page connected devices ssids are displayed, but when I try to go to third step with all connected devices selected, my mac crashes in few seconds.

**Comment by ward44120** [comment]
Dear all (ShikaSD, wootwoot1234, anonymouz4),
I am really sorry, I am a total beginner on GitHub. Does anyone could tell me the procedure to apply those recommended fix to my application JamWiFi ? Thanks all for your help, Olivier

**Comment by syddo** [comment]
I think this needs to be tested on newer Macs, ie: 2019 models.
It doesn't sound like a fix if the machine "crash" during the use this app.

**Comment by Ne7WoRK** [comment]
Is it necessary to build it from Xcode? I am really not sure how to run it. If anyone can share briefly how this is run, I would appreciate it a lot
